iOS/Flows/Conversation/Cells/MessageSequenceCollectionViewDataSource.swift

```diff
@@ -190,26 +196,36 @@ extension MessageSequenceCollectionViewDataSource: TimeMachineCollectionViewLayo
 
     func getTimeMachineItem(forItemAt indexPath: IndexPath) -> TimeMachineLayoutItemType {
         guard let item = self.itemIdentifier(for: indexPath) else {
-            return TimeMachineLayoutItem(layoutId: String())
+            return TimeMachineLayoutItem(layoutId: "", date: Date.distantPast)
         }
 
         return self.getTimeMachineItem(forItem: item)
     }
 
     private func getTimeMachineItem(forItem item: ItemType) -> TimeMachineLayoutItemType {
         switch item {
-        case .message(_, let messageID, _):
-            return TimeMachineLayoutItem(layoutId: messageID)
+        case .message(let messageId, _):
+            guard let message = self.messageSequenceController.getMessage(withId: messageId) else {
+                return TimeMachineLayoutItem(layoutId: "", date: .distantPast)
+            }
+            return TimeMachineLayoutItem(layoutId: message.id, date: message.createdAt)
         case .loadMore:
-            return TimeMachineLayoutItem(layoutId: "loadMore")
+            // Get the oldest loaded message and set the date slightly before that.
+            guard let oldestMessage = self.messageSequenceController.messageArray.first else {
+                return TimeMachineLayoutItem(layoutId: "loadMore", date: .distantPast)
+            }
+
+            return TimeMachineLayoutItem(layoutId: "loadMore",
+                                         date: oldestMessage.createdAt - 0.001)
         case .initial:
-            return TimeMachineLayoutItem(layoutId: "initial")
+            return TimeMachineLayoutItem(layoutId: "initial", date: .distantPast)
         case .placeholder:
-            return TimeMachineLayoutItem(layoutId: "placeholder")
+            return TimeMachineLayoutItem(layoutId: "placeholder", date: .distantFuture)
```

Comment:
The time machine items now include a date. This way we can figure out how to shift the collection layout when items are inserted/deleted.